### PR TITLE
feat(bootstrap): add support for AWS platform to bootstrap

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR expands the set of platforms considered as part of the bootstrap configuration process to include AWS. This will enable the population of the AWS EC2 metadata for Envoy nodes running on EC2 instances in the Envoy node metadata.

Signed-off-by: Douglas Reid <douglas-reid@users.noreply.github.com>

### CODE REVIEW

**Comment by douglas-reid** [comment]
Validated on an EKS cluster:

```
     "PLATFORM_METADATA": {
       "aws_region": "us-west-2",
       "aws_account_id": "***********4",
       "aws_instance_id": "i-***********bee4c0",
       "aws_availability_zone": "us-west-2c"
      },
...
     "locality": {
      "region": "us-west-2",
      "zone": "us-west-2c"
     },
```

**Comment by douglas-reid** [comment]
Ping for review.

**Review by rshriram — APPROVED** [review]
nice!

**Review by howardjohn — APPROVED** [review]

File: pkg/bootstrap/platform/platform.go [hunk 1]
```diff
@@ -31,3 +31,17 @@ type Environment interface {
 	// platform-specific representation into the Envoy Locality schema.
 	Locality() *core.Locality
 }
+
+// Unknown provides a default platform environment for cases in wich the platform
```
Comment (howardjohn) [inline pkg/bootstrap/platform/platform.go]:
```suggestion
// Unknown provides a default platform environment for cases in which the platform
```

**Comment by douglas-reid** [comment]
/test unit-tests_istio

flake

**Pull request merged**